cmd/shield/main.go

```diff
@@ -1188,6 +1192,36 @@ func main() {
 		r.Add("Name", t.Name)
 		r.Output(os.Stdout)
 
+	/* }}} */
+	case "delete-tenant": /* {{{ */
+		if len(args) != 1 {
+			fail(2, "Usage: shield %s [OPTIONS] NAME-or-UUID\n", command)
+		}
+		t, err := c.FindTenant(args[0], true)
+		bail(err)
+
+		if !confirm(opts.Yes, "Are you sure you want to delete all configuration under this tennant?") {
+			break
+		}
+
+		if !opts.DeleteTenant.Recursive {
+			_, err = c.DeleteTenant(t, false)
+			bail(err)
+		} else {
+			_, err = c.DeleteTenant(t, true)
+			bail(err)
+		}
```

Comment:
This whole block can just be `_, err = c.DeleteTenant(t, opts.DeleteTenant.Recursive)`